addons/fix-uploaded-svgs/userscript.js

```diff
@@ -1,4 +1,60 @@
 export default async function ({ addon, console }) {
+  function fixEditorSVG(element) {
+    const svg = element.cloneNode(true);
+
+    // Fixes the previous issue of SVGs imported from Affinity Designer
+    if (svg.height.baseVal.valueAsString === "100%" && svg.width.baseVal.valueAsString === "100%") {
+      svg.removeAttribute("height");
+      svg.removeAttribute("width");
+    }
+
+    if (addon.settings.get("removeUnsupported")) {
+      // "dominant-baseline" needs this simulated DOM to shift accordingly
+      const doc = document.createDocumentFragment();
+      doc.appendChild(svg);
```

Comment:
```suggestion
      // "dominant-baseline" needs this iframe to shift accordingly
      const iframe = document.createElement("iframe");
      iframe.setAttribute("src", "about:blank");
      iframe.setAttribute("sandbox", "allow-same-origin");
      document.body.append(iframe);
      iframe.contentDocument.body.appendChild(svg);
```

Tested this with an SVG containing `script` tags; fixes were still applied while the script execution was blocked. I'm unsure if this is sufficient